### PR TITLE
Add a test for server returning error with debug info in trailer.

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

**Comment by yang-g** [comment]
#4544

File: test/cpp/end2end/end2end_test.cc [hunk 1]
```diff
@@ -975,6 +975,31 @@ TEST_P(End2endTest, NonExistingService) {
...
+  grpc::string expected_string = info->SerializeAsString();
+  request.set_message("Hello");
+
+  Status s = stub_->Echo(&context, request, &response);
+  EXPECT_FALSE(s.ok());
+  auto trailers = context.GetServerTrailingMetadata();
+  EXPECT_EQ(1, trailers.count(kDebugInfoTrailerKey));
+  auto iter = trailers.find(kDebugInfoTrailerKey);
+  EXPECT_EQ(expected_string, iter->second);
+}
```
Comment (jcanizales) [inline test/cpp/end2end/end2end_test.cc]: As the goal is mostly to be useful to people as sample code, could you add parsing the trailer back into proto message form?

Comment (yang-g) [inline test/cpp/end2end/end2end_test.cc]: Done.

**Comment by yang-g** [comment]
PTAL

**Comment by jcanizales** [comment]
The error in `concurrent_connectivity_test` on Windows seems to be a flake that's not currently tracked. Do you mind opening an issue for it and pointing to this PR from it? Apart from that LGTM.

**Comment by yang-g** [comment]
Added a note to #6319, which is the blanket issue for windows flakes.

**Pull request merged**